designer/server/src/common/components/pagination/template.test.js

```diff
@@ -0,0 +1,580 @@
+import { renderMacro } from '~/test/helpers/component-helpers.js'
+
+describe('Pagination Component', () => {
+  /** @type {Element | null} */
+  let $pagination = null
+
+  describe('With pagination data', () => {
+    describe('Standard pagination', () => {
+      beforeEach(() => {
+        renderMacro('appPagination', 'pagination/macro.njk', {
+          params: {
+            baseUrl: '/library',
+            pagination: {
+              page: 2,
+              perPage: 10,
+              totalPages: 5,
+              totalItems: 50,
+              pages: [
+                {
+                  number: '1',
+                  href: '/library?page=1&perPage=10',
+                  current: false
+                },
+                {
+                  number: '2',
+                  href: '/library?page=2&perPage=10',
+                  current: true
+                },
+                {
+                  number: '3',
+                  href: '/library?page=3&perPage=10',
+                  current: false
+                },
+                { ellipsis: true },
+                {
+                  number: '5',
+                  href: '/library?page=5&perPage=10',
+                  current: false
+                }
+              ]
+            }
+          }
+        })
+
+        $pagination = document.querySelector('.govuk-pagination')
+      })
+
+      it('should render the pagination component', () => {
+        expect($pagination).toBeInTheDocument()
+      })
+
+      it('should have the correct previous and next links', () => {
+        const $previousLink = $pagination?.querySelector(
+          '.govuk-pagination__prev a'
+        )
+        const $nextLink = $pagination?.querySelector(
+          '.govuk-pagination__next a'
+        )
+
+        expect($previousLink).toBeInTheDocument()
+        expect($previousLink).toHaveAttribute(
+          'href',
+          '/library?page=1&perPage=10'
+        )
+        expect($previousLink).toHaveTextContent('Previous')
+
+        expect($nextLink).toBeInTheDocument()
+        expect($nextLink).toHaveAttribute('href', '/library?page=3&perPage=10')
+        expect($nextLink).toHaveTextContent('Next')
+      })
+
+      it('should render the correct page numbers', () => {
+        const $pages = $pagination?.querySelectorAll('.govuk-pagination__item')
+        expect($pages).toHaveLength(5)
+
+        expect($pages?.[0].textContent?.trim()).toBe('1')
```

Comment:
Love seing all these Do the DOM assertions do the trim bit for us?

```suggestion
        expect($pages?.[0]).toHaveTextContent('1')
```